src/wallet/wallet.cpp

```diff
@@ -4431,7 +4480,7 @@ void CWallet::GetFilteredNotes(std::vector<CNotePlaintextEntry> & outEntries, st
             PaymentAddress pa = nd.address;
 
             // skip notes which belong to a different payment address in the wallet
-            if (fFilterAddress && !(pa == filterPaymentAddress)) {
+            if (!(filterAddresses.empty() || filterAddresses.count(pa))) {
```

Comment:
Seems to me a bit convoluted to read. What is the intention here ? 
Due to the negation it does not continue when:
- either `filterAddresses.empty() == true` (which implies `filterAddresses.count(pa) == 0` for sure)
- or `filterAddresses.count(pa) != 0` (which implies `filterAddresses.empty() == false` for sure)

It continues only if `filterAddresses.count(pa) == 0` or in other word if `pa ∉ filterAddesses`

So why not simply ?
```suggestion
            if (!filterAddresses.contains(pa)) {
```

Reply:
Which raises an immediate question ... 
why even enter the loop `for (auto & pair : wtx.mapNoteData) {` if `filterAddresses` is empty ?
Under such condition there is no point traversing it and do continuous lookups on an empty set

Reply:
I agree the condition is a bit convoluted to read, but I believe your suggestion is not logically equal.
With the original loop, if `filterAddresses` is empty you do not execute `continue`, while with your suggestion you do execute `continue`.

I do not have the full picture clear yet, but my guess is that `filterAddresses` is an optional whitelist that you might not use. If that is the case, my suggestion would be:

```suggestion
            if (!filterAddresses.empty() && !filterAddresses.contains(pa)) {
```